packages/beacon-node/src/sync/unknownBlock.ts

```diff
@@ -304,130 +332,69 @@ export class UnknownBlockSync {
     block.status = PendingBlockStatus.fetching;
 
     let res: Result<{blockInput: BlockInput; peerIdStr: string}>;
-    let connectedPeers: string[];
     if (block.blockInput === null) {
-      connectedPeers = allPeers;
       // we only have block root, and nothing else
-      res = await wrapError(this.fetchUnknownBlockRoot(fromHex(block.blockRootHex), connectedPeers));
+      res = await wrapError(this.fetchUnknownBlockRoot(fromHex(block.blockRootHex)));
     } else {
-      const {cachedData} = block.blockInput;
-      if (cachedData.fork === ForkName.fulu) {
-        const {dataColumnsCache} = cachedData as CachedDataColumns;
-        const sampledColumns = this.network.custodyConfig.sampledColumns;
-        const neededColumns = sampledColumns.reduce((acc, elem) => {
-          if (dataColumnsCache.get(elem) === undefined) {
-            acc.push(elem);
-          }
-          return acc;
-        }, [] as number[]);
-
-        connectedPeers =
-          neededColumns.length <= 0
-            ? allPeers
-            : allPeers.filter((peer) => {
-                const {custodyGroups: peerColumns} = this.network.getConnectedPeerSyncMeta(peer);
-                const columns = peerColumns.reduce((acc, elem) => {
-                  if (neededColumns.includes(elem)) {
-                    acc.push(elem);
-                  }
-                  return acc;
-                }, [] as number[]);
-                return columns.length > 0;
-              });
-        if (connectedPeers.length > 0) {
-          this.logger.debug("Filtered peers to those having relevant columns for downloading data", {
-            ...logCtx,
-            allPeers: allPeers.length,
-            connectedPeers: connectedPeers.length,
-          });
-        } else {
-          this.logger.debug("Skipping download as no filtered peers having relevant data", {
-            ...logCtx,
-            allPeers: allPeers.length,
-            connectedPeers: connectedPeers.length,
-            neededColumns: neededColumns.join(" "),
-          });
-          return;
-        }
-      } else {
-        connectedPeers = allPeers;
-      }
-      res = await wrapError(this.fetchUnavailableBlockInput(block.blockInput, connectedPeers));
+      res = await wrapError(this.fetchUnavailableBlockInput(block.blockInput));
     }
 
     if (res.err) this.metrics?.syncUnknownBlock.downloadedBlocksError.inc();
     else this.metrics?.syncUnknownBlock.downloadedBlocksSuccess.inc();
 
     if (!res.err) {
       const {blockInput, peerIdStr} = res.result;
+      // fetchUnknownBlockRoot and fetchUnavailableBlockInput should return available data BlockInput, throw error if not
       if (blockInput.type === BlockInputType.dataPromise) {
```

Comment:
Should this check for available data?  `fetchUnavailableBlockInput` can return for non-`NullBlockInput` and non-data promise.  Might be safer to check for the inverse of the desired condition instead.

```suggestion
      if (blockInput.type !== BlockInputType.availableData) {
```